### PR TITLE
Attach pointcloudmodel opc

### PR DESCRIPTION
**Pull request by LeePipBull**
Adds point cloud support for attached OPC point cloud models, equivalent to existing TileTree point cloud logic.
Resolves RDS URL to OrbitGTBlobProps for Blob URL and SAS token.

Creation of iModel with OPC PointCloudModel attachment currently for testing only and requires the following entry in ImportConfig.xml in the Microstation Connector assets:

<OptionString name="PointCloudOutputFormat" value="OPC"/>

### CODE REVIEW

File: common/api/ecschema-metadata.api.md [hunk 1]
```diff
@@ -420,7 +420,11 @@ export const Diagnostics: {
             readonly diagnosticType: DiagnosticType;
             ecDefinition: Schema;
             messageArgs?: [string, string] | undefined;
-            category: DiagnosticCategory;
+            category: DiagnosticCategory; /**
```
Comment (pmconne) [inline common/api/ecschema-metadata.api.md:423]: I think you need to revert the changes to ecschema-metadata.api.md. extract-api behaves funkily with this package for some reason.

File: core/common/src/ContextRealityModel.ts [hunk 2]
```diff
@@ -15,6 +15,7 @@ import { FeatureAppearance, FeatureAppearanceProps } from "./FeatureSymbology";
  * @alpha
  */
 export interface OrbitGtBlobProps {
+  rdsUrl: string;
```
Comment (pmconne) [inline core/common/src/ContextRealityModel.ts:18]: You have added a new property, which will not be present in older data; therefore it needs to be marked optional (`rdsUrl?: string`).

**Review by pmconne — CHANGES_REQUESTED** [review]

File: common/api/ecschema-metadata.api.md (hunk 1 above)
Comment (LeePipBull) [reply common/api/ecschema-metadata.api.md:423]: Thanks. I actually saw it and didn't initially commit it as it didn't appear to have any affect other than comments and it seemed suspicious as to why it hadn't already been committed with them.

However, the build is complaining about this file, so I tried including it. I'll push a revert now.

File: core/common/src/ContextRealityModel.ts (hunk 2 above)
Comment (LeePipBull) [reply core/common/src/ContextRealityModel.ts:18]: Sure, thanks, I'll include that.

File: core/frontend/src/tile/OrbitGtTileTree.ts [hunk 3]
```diff
@@ -359,6 +359,10 @@ export namespace OrbitGtTileTree {
 
   async function initializeBlobStorage(_props: OrbitGtBlobProps, iModel: IModelConnection, _modelId: Id64String) {
 
+    // rdsUrl must be defined and not null to be resolved
+    if(_props.rdsUrl == null)
```
Comment (pmconne) [inline core/frontend/src/tile/OrbitGtTileTree.ts:363]: The linter's going to complain about `==` vs `===`, and `===` won't catch `undefined`.
If you would consider an empty string also unresolvable, consider replacing with `if (!_props.rdsUrl)`.

File: core/common/src/ContextRealityModel.ts (hunk 2 above)
Comment (LeePipBull) [reply core/common/src/ContextRealityModel.ts]: I've made rdsUrl an option rdsUrl?

File: core/frontend/src/ModelState.ts [hunk 4]
```diff
@@ -113,15 +113,36 @@ export abstract class GeometricModelState extends ModelState implements Geometri
...
   public createTileTreeReference(view: ViewState): TileTreeReference {
     // If this is a reality model, its tile tree is obtained from reality data service URL.
+
     const url = this.jsonProperties.tilesetUrl;
     if (undefined !== url) {
       const spatialModel = this.asSpatialModel;
-      return createRealityTileTreeReference({
-        url,
+
+      if (url.orbitGtBlob === undefined) {
```
Comment (pmconne) [inline core/frontend/src/ModelState.ts:121]: Up until now a GeometricModel's `tilesetUrl` JSON property was always a string. Existing code will potentially break if it now encounters an object instead of a string. Did you consider adding a different JSON property instead of making the existing `tilesetUrl` property serve dual duty?

File: core/frontend/src/tile/OrbitGtTileTree.ts (hunk 3 above)
Comment (LeePipBull) [reply core/frontend/src/tile/OrbitGtTileTree.ts]: I've changed it to !rdsUrl, although it wasn't causing an issue in the linter locally, I suspected it was configured to permit == with null.

File: core/frontend/src/ModelState.ts (hunk 4 above)
Comment (LeePipBull) [reply core/frontend/src/ModelState.ts:121]: tilesetUrl was already either a URL to a tileset JSON root document if the point cloud was a TileTree, or an OrbitGtBlobProps object if it's an OPC format point cloud. 

Here's an existing example: https://github.com/imodeljs/imodeljs/blob/master/core/frontend/src/ContextRealityModelState.ts#:~:text=/**%20%40internal%20*/,%7D)%20%3A

We are not keen on it being this way and are discussing using an alternative data structure in future.

Comment (LeePipBull) [reply core/frontend/src/ModelState.ts:121]: Thanks, I'm going to review that. I thought the decision to do that was based on some existing code taking that approach, but looking back I'm not able to see it. It would make far more sense to have it separate. We've been discussing simplifying the ways reality data is referenced anyway.

Comment (LeePipBull) [reply core/frontend/src/ModelState.ts]: I've updated the orbitGtBlob to be a separate property (and not reuse tilsetUrl).
There is a PR for MstnBridge on iModel02 for  that creates the iModel with this alteration.

**Review by pmconne — APPROVED** [review]
